ui/app.py

```diff
@@ -1,127 +1,112 @@
 import openai
 import streamlit as st
-from typing import List
+from typing import Optional, Callable, Union
 
 from rip.utils import initialize_openai_api, DEFAULT_IMG_URL
 from ui.components.input_data import input_data
-from rip.utils import get_default_completion_params, set_completion_params
+from ui.components.image_generation import generate_image
+from ui.components.text_completion import (
+    get_default_completion_params,
+    transform_text_to_scene_description,
+    get_semantic_search_results,
+    get_suggested_actions
+)
+
+DEFAULT_COMPLETION_PARAMS = get_default_completion_params()
 
 
 def init():
     # page settings
     st.set_page_config(
-        layout="centered",        
+        layout="centered",
         page_title="Reveal Impacts of eco-friendly Policies (RIP)",
         page_icon="eco",
         menu_items={"Get help": "https://github.com/Ohyeon5/RIP"},
     )
     # initialize openai api's variables
     initialize_openai_api()
 
+
 def get_search_question():
-    #* Finding resources we can use as an input for our solution. (GPT3 Compare for semantic search and recommendations)
+    # * Finding resources we can use as an input for our solution. (GPT3 Compare for semantic search and recommendations)
     st.subheader("What solution you are thinking about against lowering CO2 emission?")
     question = st.text_input('What is your favorite solution:', "I'll purchase an electric car")
     return question
 
-def return_search_results(results: List[str]):
-    #* Grouping, filtering and transforming the inputs which migh be useful for our search on second layer affects of policies and applications. (GPT3 Edit)
+
+def return_search_results(question):
+    results = get_semantic_search_results(question)
+    # * Grouping, filtering and transforming the inputs which migh be useful for our search on second layer affects of policies and applications. (GPT3 Edit)
     st.subheader("We found those side affects regarding your choice")
-    st.write(", ".join(results))
-
-def suggest_actions(actions: List[str]):
-    #* Generating quotes and new short definitions from what we found, as a list of takeaways. (GPT3 Explain and Write)
-    st.subheader("List of takeaways which might be considered, according to current policies")
-    action = st.radio(
-            "Which action best fits?",
-            actions)
-    dalle2(action)
-  
-    
-def tldr():     
+    st.write(results)
+    actions = get_suggested_actions(question, results)
+    return actions
+
+def suggest_actions(actions: str):
+    # * Generating quotes and new short definitions from what we found, as a list of takeaways. (GPT3 Explain and Write)
+    st.subheader(
+        "List of takeaways which might be considered"
```

Comment:
suggestions, something like "List of takeaways you might consider:" or "Here are a list of takeaways I would consider:"